File: tags/plugin-pac-1.0.1/src/java/fr/paris/lutece/plugins/pac/dao/artifact/ArtifactDAO.java
package fr.paris.lutece.plugins.pac.dao.artifact;

import fr.paris.lutece.plugins.pac.bean.artifact.Artifact;
import fr.paris.lutece.plugins.pac.dao.AbstractPacDAO;

public class ArtifactDAO extends AbstractPacDAO<Integer, Artifact> implements IArtifactDAO
{

    @Override
    protected Class<Artifact> getBeanClass( )
    {
        return Artifact.class;
    }

}

File: trunk/plugin-scala/src/main/scala/fr/paris/lutece/plugins/scala/dao/ArtifactDAO.scala
package fr.paris.lutece.plugins.scala.dao

import fr.paris.lutece.plugins.genericjpa.dao.AbstractDAO
import fr.paris.lutece.plugins.scala.bean.Artifact

class ArtifactDAO extends AbstractDAO[Int, Artifact]{

  def getBeanClass():Class[Artifact]={
    classOf[Artifact]
  }
  
  def getPluginName():String={
    "scala"
  }
  
  def findByStrPrimaryKey(key:String):Artifact={
    null
  }
}